src/features/bivariate_color_manager/components/ColorLegendFilters/ColorLegendFilters.tsx

```diff
@@ -0,0 +1,231 @@
+import { Autocomplete, Select } from '@konturio/ui-kit';
+import { useCallback, useMemo, useRef, useState } from 'react';
+import { useAtom } from '@reatom/react';
+import { Plus16 } from '@konturio/default-icons';
+import { nanoid } from 'nanoid';
+import cn from 'clsx';
+import { i18n } from '~core/localization';
+import { bivariateColorManagerResourceAtom } from '~features/bivariate_color_manager/atoms/bivariateColorManagerResource';
+import { capitalize } from '~utils/common';
+import { bivariateColorManagerDataAtom } from '~features/bivariate_color_manager/atoms/bivariateColorManagerData';
+import style from './ColorLegendFilters.module.css';
+import type { SelectItemType, AutocompleteItemType } from '@konturio/ui-kit';
+
+const SentimentFiltersClasses = {
+  noValue: style.NoValue,
+};
+
+const LayerFilterClasses = {
+  menu: style.AutocompleteMenu,
+};
+
+type FilterValueType = { key: string; value: string[] };
+
+export const ColorLegendFilters = () => {
+  const [indicators, { setLayersFilter, setSentimentsFilter }] = useAtom(
+    bivariateColorManagerDataAtom,
+    (state) => state.indicators,
+  );
+
+  const [loading] = useAtom(
+    bivariateColorManagerResourceAtom,
+    (state) => state.loading,
+  );
+
+  const selectIndicatorsData: SelectItemType[] = useMemo(() => {
+    return (
+      indicators?.map((indicator) => ({
+        title: indicator.label,
+        value: indicator.name,
+      })) || []
+    );
+  }, [indicators]);
+
+  // calculate unique direction items that are available in indicators list
+  const selectDirectionsData: SelectItemType[] = useMemo(() => {
+    const uniqueDirectionsSet = new Set<string>();
+    const directionsData: SelectItemType[] = [];
+
+    function processDirectionItem(directionItem: string) {
+      if (!uniqueDirectionsSet.has(directionItem)) {
```

Comment:
it feels like instead of Set + Array you can just use Map to collect unique keys.
in the end you can [...map.values()] , it will guarantee insertion order